dev/update_changelog.py

```diff
@@ -0,0 +1,173 @@
+import re
+import os
+import subprocess
+from datetime import datetime
+from collections import defaultdict
+from pathlib import Path
+from typing import NamedTuple, List, Any
+
+import click
+import requests
+
+
+def get_header_for_version(version):
+    return "## {} ({})".format(version, datetime.now().strftime("%Y-%m-%d"))
+
+
+def extract_pr_num_from_git_log_entry(git_log_entry):
+    m = re.search(r"\(#(\d+)\)$", git_log_entry)
+    return int(m.group(1)) if m else None
+
+
+class PullRequest(NamedTuple):
+    title: str
+    number: int
+    author: str
+    labels: List[str]
+
+    @property
+    def url(self):
+        return f"https://github.com/mlflow/mlflow/pull/{self.number}"
+
+    @property
+    def release_note_labels(self):
+        return [l for l in self.labels if l.startswith("rn/")]
+
+    def __str__(self):
+        return f"{self.title} (#{self.number}, @{self.author})"
+
+    def __repr__(self):
+        return str(self)
+
+
+class Section(NamedTuple):
+    title: str
+    items: List[Any]
+
+    def __str__(self):
+        if not self.items:
+            return ""
+        return "\n\n".join(
+            [
+                self.title,
+                "\n".join(f"- {item}" for item in self.items),
+            ]
+        )
+
+
+@click.command(help="Update CHANGELOG.md")
+@click.option(
+    "--prev-branch",
+    required=True,
+    help="Previous release branch to compare to, e.g. branch-0.8",
+)
+@click.option(
+    "--curr-branch",
+    default="master",
+    help="Current release (candidate) branch to compare to, e.g. branch-0.9 (default: 'master').",
+)
+@click.option("--release-version", required=True, help=("MLflow version to release."))
+def main(prev_branch, curr_branch, release_version):
```

Comment:
The original script (in our internal repository) updates `CHANGELOG.md` and files a PR, but this script just updates `CHANGELOG.md`.